browser/src/control/Control.Toolbar.js

```diff
@@ -1404,7 +1404,14 @@ function setupToolbar(e) {
 				//click pos tweak
 				cellCursor._northEast.lng = cellCursor._southWest.lng;
 				map._docLayer._closeURLPopUp();
-				map._docLayer._showURLPopUp(cellCursor._northEast, e.url);
+				var linkPosition;
+				if (strTwips.length > 7) {
+					linkPosition = {
+						x: parseInt(strTwips[6]),
+						y: parseInt(strTwips[7])
```

Comment:
Looks OK, but this gets very complicated at this stage, could you add a comment above this block and describe the purpose of the 8 numbers you get here as a parameter?

Also, we have a cypress test suite at cypress_test/integration_tests/desktop/calc/clipboard_spec.js, do you think you could cover your fix with a test there? Thanks.